Makefile

```diff
@@ -16,19 +16,21 @@ deps: vndr
 	go get -u $(BUILDFLAGS) github.com/golang/lint/golint
 	go get $(BUILDFLAGS) github.com/vbatts/git-validation
 
-vndr:
-	go get -u github.com/LK4D4/vndr
-	vndr
+vndr: vendor.conf
+	@go get -u github.com/LK4D4/vndr && vndr
+
+clean:
+	rm -rf vendor
 
 test: deps
-	 @go test $(BUILDFLAGS) -cover $(PACKAGES)
+	@go test $(BUILDFLAGS) -cover $(PACKAGES)
 
 # This is not run as part of (make all), but Travis CI does run this.
 # Demonstarting a working version of skopeo (possibly with modified SKOPEO_REPO/SKOPEO_BRANCH, e.g.
 #    make test-skopeo SKOPEO_REPO=runcom/skopeo-1 SKOPEO_BRANCH=oci-3 SUDO=sudo
 # ) is a requirement before merging; note that Travis will only test
 # the master branch of the upstream repo.
-test-skopeo:
+test-skopeo: clean
```

Comment:
It is _extremely_ non-obvious why the `clean` here matters, and it is unintuitive that `make test-something` modifies the vendored libraries in the current directory.

At the very least this needs a detailed comment explaining why the “clean” here is essential and shouldn’t be removed; but I’d much prefer dropping this dependency, and doing `rm -rf $${vendor_path} && cp -r . $${vendor_path}`**` && rm -rf $${vendor_path}/vendor`** below.